assignment-6.cpp

```diff
@@ -0,0 +1,81 @@
+#include <iostream>
+#include <vector>
+#include <map>
+using namespace std;
+struct moves {
+    int start, dest;
+};
+
+void print_moves(vector<moves> mov) {
+    for (int i = 0; i < mov.size(); i++)
+        cout << "move from " << mov[i].start << " to " << mov[i].dest << endl;
+}
+
+vector<int> swap_cars(vector<int> current_state, vector<int> final_state, map<int, int> *car_positions, int car1, int car2, int now_fixing) {
```

Comment:
I think since you don't change final_state anywhere in your code, it is better to be passed as a const. :)